src/Microsoft.AspNet.Server.Kestrel/Http/SocketOutput.cs

```diff
@@ -46,21 +44,26 @@ public void Write(ArraySegment<byte> buffer, Action<Exception, object> callback,
         public class ThisWriteReq : UvWriteReq
         {
             private static readonly Action<UvWriteReq, int, Exception, object> _writeCallback = WriteCallback;
+
             private static void WriteCallback(UvWriteReq req, int status, Exception error, object state)
             {
                 ((ThisWriteReq)state).OnWrite(req, status, error);
             }
 
             SocketOutput _self;
-            ArraySegment<byte> _buffer;
+            byte[] _buffer;
```

Comment:
That part of the server hasn't been optimized at all really (which is why it's copying the array segment every time) - so I wouldn't worry too much about fixing that code path before it's been fine-tuned.